v2/googlecloud-to-googlecloud/src/main/java/com/google/cloud/teleport/v2/templates/bigtablechangestreamstopubsub/schemautils/PubSubUtils.java

```diff
@@ -0,0 +1,396 @@
+/*
+ * Copyright (C) 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.cloud.teleport.v2.templates.bigtablechangestreamstopubsub.schemautils;
+
+import com.google.cloud.pubsub.v1.Publisher;
+import com.google.cloud.teleport.bigtable.ChangelogEntryMessage;
+import com.google.cloud.teleport.bigtable.ChangelogEntryMessageJson;
+import com.google.cloud.teleport.bigtable.ChangelogEntryMessageProto;
+import com.google.cloud.teleport.bigtable.ModType;
+import com.google.cloud.teleport.v2.templates.bigtablechangestreamstopubsub.model.BigtableSource;
+import com.google.cloud.teleport.v2.templates.bigtablechangestreamstopubsub.model.PubSubDestination;
+import com.google.cloud.teleport.v2.templates.bigtablechangestreamstopubsub.model.PubSubFields;
+import com.google.protobuf.ByteString;
+import com.google.protobuf.util.JsonFormat;
+import com.google.pubsub.v1.PubsubMessage;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.Serializable;
+import java.nio.ByteBuffer;
+import java.nio.charset.Charset;
+import java.util.Base64;
+import java.util.EnumMap;
+import org.apache.avro.io.Encoder;
+import org.apache.avro.io.EncoderFactory;
+import org.apache.commons.lang3.Validate;
+import org.json.JSONObject;
+
+/**
+ * {@link PubSubUtils} provides utils for processing PubSub schema and generating PubSub messages.
+ */
+public class PubSubUtils implements Serializable {
+
+  private static final EnumMap<PubSubFields, PubSubValueFormatter> FORMATTERS =
+      new EnumMap<>(PubSubFields.class);
+
+  static {
+    FORMATTERS.put(
+        PubSubFields.ROW_KEY_STRING,
+        (pb, chg) -> {
+          String rowkeyEncoded = chg.getString(PubSubFields.ROW_KEY_BYTES.name());
+          return pb.convertBase64ToString(rowkeyEncoded);
+        });
+    FORMATTERS.put(
+        PubSubFields.ROW_KEY_STRING_BASE64, (pb, chg) -> PubSubFields.ROW_KEY_STRING_BASE64);
+    FORMATTERS.put(
+        PubSubFields.ROW_KEY_BYTES,
+        (pb, chg) -> {
+          String rowkeyEncoded = chg.getString(PubSubFields.ROW_KEY_BYTES.name());
+          return pb.convertBase64ToBytes(rowkeyEncoded);
+        });
+    FORMATTERS.put(PubSubFields.MOD_TYPE, (pb, chg) -> chg.getString(PubSubFields.MOD_TYPE.name()));
+    FORMATTERS.put(
+        PubSubFields.COMMIT_TIMESTAMP,
```

Comment:
I think we should rename COMMIT_TIMESTAMP into COMMIT_TIMESTAMP_NUM, because we only use numeric timestamps. Also the implementation will throw NumberFormatException at like 68 because it effectively tries to do this:

`Long.parseLong("COLUMN_FAMILY");`